### PR TITLE
[Codemagic] Add Codemagic extension

### PR DESCRIPTION
**Pull request by rrroyal**
## Description
This extension implements integration with [Codemagic](https://codemagic.io), a CI/CD platform. It allows users to see all of their configured apps and latest builds.

## Screencast
Builds List Screenshot

## Checklist
- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/codemagic/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Codemagic
+See your Codemagic apps right in Raycast!
+
+# Setup
+Grab your token from [Codemagic settings](https://codemagic.io/settings) and you're good to go!
```
Comment (pernielsentikaer) [inline extensions/codemagic/README.md:5]: It seems like it's better to tell users to do `Teams > Personal Account > Integrations > Codemagic API > Show` - the page you're linking is not the right place 🫤

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

* Adding an [List.EmptyView](https://developers.raycast.com/api-reference/user-interface/list#list.emptyview) when the token is wrong, the current error handling is not possible for the end user to understand what's wrong

image

* I can't get add/remove favorites to work; nothing happens 

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/codemagic/README.md (hunk 1 above)
Comment (rrroyal) [reply extensions/codemagic/README.md:5]: Oh, you're right! They must've changed it :/

**Comment by pernielsentikaer** [comment]
Are you still around @rrroyal

**Pull request closed without merging**